File: src/test/java/io/binac/leetcode/BestTimeToBuyAndSellStockTest.java
package io.binac.leetcode;

import org.junit.jupiter.api.Test;

import static org.junit.jupiter.api.Assertions.*;

class BestTimeToBuyAndSellStockTest {
    private final BestTimeToBuyAndSellStock.Solution1 solution1 = new BestTimeToBuyAndSellStock.Solution1();

    @Test
    void test1() {
        int prices1[] = {7, 1, 5, 3, 6, 4};
        assertEquals(5, solution1.maxProfit(prices1));

        int prices2[] = {7, 6, 4, 3, 1};
        assertEquals(0, solution1.maxProfit(prices2));
    }
}

File: cpp/tests/BestTimeToBuyAndSellStockTest.cpp
#include "BestTimeToBuyAndSellStock.h"
#include "gtest/gtest.h"

using namespace lcpp;

TEST(BestTimeToBuyAndSellStock, Solution121_1) {
  auto S121_1 = Solution121_1();

  std::vector<int> Prices1{7, 1, 5, 3, 6, 4};
  int const Expected1 = 5;
  EXPECT_EQ(Expected1, S121_1.maxProfit(Prices1));

  std::vector<int> Prices2{7, 6, 4, 3, 1};
  int const Expected2 = 0;
  EXPECT_EQ(Expected2, S121_1.maxProfit(Prices2));
}